app/models/patient.rb

```diff
@@ -795,16 +828,22 @@ def report_eligibility
 
     # Rough estimate of next contact time
     if eligible
-      messages << case preferred_contact_time
-                  when 'Morning'
-                    { message: '8:00 AM local time (Morning)', datetime: nil }
-                  when 'Afternoon'
-                    { message: '12:00 PM local time (Afternoon)', datetime: nil }
-                  when 'Evening'
-                    { message: '4:00 PM local time (Evening)', datetime: nil }
-                  else
-                    { message: 'Today', datetime: nil }
-                  end
+      local_time = Time.now.getlocal(self.time_zone_offset)
+      hour_in_timezone = local_time.hour
+      case preferred_contact_time
+      when 'Morning'
+        eligible = false unless hour_in_timezone >= 8 && hour_in_timezone <= 11
+        messages << { message: "Monitoree should be contacted between 8:00 AM and 12:00 PM local time (Morning). Current local time: #{local_time.strftime("%I:%M %p")}"}
+      when 'Afternoon'
+        eligible = false unless hour_in_timezone >= 12 && hour_in_timezone <= 15
+        messages << { message: "Monitoree should be contacted between 12:00 PM and 4:00 PM local time (Afternoon). Current local time: #{local_time.strftime("%I:%M %p")}"}
+      when 'Evening'
+        eligible = false unless hour_in_timezone >= 16 && hour_in_timezone <= 18
+        messages << { message: "Monitoree should be contacted between 4:00 PM and 7:00 PM local time (Evening). Current local time: #{local_time.strftime("%I:%M %p")}"}
+      else
+        eligible = false unless hour_in_timezone >= 11 && hour_in_timezone <= 16
+        messages << { message: "Monitoree should be contacted between 11:00 AM and 5:00 PM local time. Current local time: #{local_time.strftime("%I:%M %p")}"}
+      end
     end
```

Comment:
Does this language change need to be approved/reviewed elsewhere? @Bialogs @holmesie

Reply:
Why did the language and logic have to change?

Reply:
The logic for eligibility was inconsistent with how we discussed the report_eligibility scope should work. The language was changed because the previous language would not reasonably explain why a patient was not eligible for notifications at a specific time of day outside of the preferred contact time window in their local time. The new language actually allows a user to see the patient's preferred contact time window as well as the current local time for that patient. I can change this back if these justifications are not acceptable.

Reply:
Okay I am good with the rationale for the changes but the PO team needs to be made aware of these changes to the linelist language. Asking for feedback in the #saraalert-ui channel is probably your best bet (after explaining the rationale for the changes)

I also would suggest updating the language for `Current local time:` to be explicit about the subject of the current local time, such as 

`"Monitoree should be contacted between 11:00 AM and 5:00 PM local time; their current local time is #{local_time.strftime("%I:%M %p")}"}`

Reply:
@Bialogs Made that language change

Reply:
Yep I also see that you asked in the channel so I'll resolve this just because the feedback on this bit of code won't be coming from this channel of communication anymore.